core/src/main/java/co/cask/wrangler/steps/transformation/functions/GeoFences.java

```diff
@@ -0,0 +1,134 @@
+/*
+ *
+ *  * Copyright © 2017 Cask Data, Inc.
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ *  * use this file except in compliance with the License. You may obtain a copy of
+ *  * the License at
+ *  *
+ *  * http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  * License for the specific language governing permissions and limitations under
+ *  * the License.
+ *
+ */
+
+package co.cask.wrangler.steps.transformation.functions;
+
+import com.github.filosganga.geogson.gson.GeometryAdapterFactory;
+import com.github.filosganga.geogson.model.Coordinates;
+import com.github.filosganga.geogson.model.Feature;
+import com.github.filosganga.geogson.model.FeatureCollection;
+import com.github.filosganga.geogson.model.positions.LinearPositions;
+import com.github.filosganga.geogson.model.positions.Positions;
+import com.github.filosganga.geogson.model.positions.SinglePosition;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * GeoFencing check based on location and polygon
+ */
+public class GeoFences {
+
+  /**
+   * Static method to be used with jexl
+   *
+   * @param latitude
+   * @param longitude
+   * @param geofence
+   * @return
+   */
+  public static final Boolean evaluate(double latitude, double longitude, String geofence) {
+    Coordinates location = Coordinates.of(longitude, latitude);
+    Boolean inzone = false;
+    Gson gson = new GsonBuilder()
+        .registerTypeAdapterFactory(new GeometryAdapterFactory())
+        .create();
+
+    FeatureCollection featureCollection = gson.fromJson(geofence, FeatureCollection.class);
+    for (Feature feature : featureCollection.features()) {
+      inzone = inzone || isPointInside(getPolygonAsList(feature), location);
+
+    }
+    return inzone;
+  }
+
+  public static final Boolean evaluate(String latitude, String longitude, String geofence) {
+    return evaluate(Double.parseDouble(latitude), Double.parseDouble(longitude), geofence);
```

Comment:
What happens when it throws exception ? Is there a proper message being returned ? PleSe handle error messages correctly